netcore/System.Private.CoreLib/src/System/Array.cs

```diff
@@ -429,12 +429,14 @@ public int GetUpperBound (int dimension)
 			return GetLowerBound (dimension) + GetLength (dimension) - 1;
 		}
 
+		// This is intrinsic.
 		[MethodImpl (MethodImplOptions.AggressiveInlining)]
 		internal ref byte GetRawSzArrayData ()
 		{
 			return ref Unsafe.As<RawData>(this).Data;
 		}
 
+		// This is intrinsic.
```

Comment:
There is a special attribute for it: `[Intrinsic]`

Reply:
We seem to have far more intrinsics than `[Intrinsic]`s.
I just keep tripping over them, figured I should start documenting.
i.e. I had changed the signature here -- crash.

Reply:
Please use it (so we can use it by other tools)

Reply:
It is just a hint to humans, right?
I mean, if I use it accidentally too much or too little, no harm done, no big concern about correctness?

I guess we could actually try to get it correct, and speed up all the checks as a result.

It almost seems like, hint to linker -- remove this IL. But I don't think that ends up working. Unless maybe reflection is also removed? Is that the idea? Like, actually more than a hint, but significant behavior?

Reply:
@jaykrell Currently it's just a hint but I guess we indeed can use them to speed up checks. 👍

Reply:
yes, we plan to use it as an optional hint for linker. For the rest check #8404

Reply:
the only problem:
Some `[Intrinsics]` are CoreCLR only, some are Mono only, some are LLVM only (e.g. `Buffer.Memmove` lives in coreclr/shared and it's only [intrinsic] for Mono-LLVM, etc.

Reply:
That's fine, the attribute should be used when it's intrinsic on most configurations